doc/docusaurus/docs/auction-smart-contract/end-to-end/generating-keys.md

```diff
@@ -0,0 +1,44 @@
+---
+sidebar_position: 5
+---
+
+# Generating Keys and Addresses
+
+To start, clone the plutus-tx-template repository into the `on-chain` directory.
+Then, create a separate `off-chain` directory, set up `package.json` and add the required dependencies:
+
+```
+git clone git@github.com:IntersectMBO/plutus-tx-template.git on-chain
+mkdir off-chain && cd $_
+yarn init -y
```

Comment:
This line implies that `yarn` is installed and available in $PATH.
Would it be better to be explicit about such per-requisites?

Reply:
I would put it in the nix shell rather than instructing users to install it, because otherwise we would also need instructions for installing GHC, Cabal etc.

Reply:
That's what I thought too. Personally I like the idea, but also nix isn't everyone's cup of tea...

Reply:
Yeah, OK, on a second thought I also don't like the idea of creating more nix flakes. I'll add the instruction for installing yarn etc.